realm/src/main/java/io/realm/internal/TableView.java

```diff
@@ -73,19 +73,24 @@
      * @param parent A table.
      * @param nativePtr pointer to table.
      */
-    protected TableView(Context context, Table parent, long nativePtr){
+    protected TableView(Context context, Table parent, long nativePtr) {
         this.context = context;
         this.parent = parent;
         this.nativePtr = nativePtr;
     }
 
+    protected TableView(Context context, Table parent, long nativePtr, TableQuery query) {
```

Comment:
The TableQuery object might hold a reference, but I believe we need to have a reference to the TableQuery object in order to prevent premature deallocation.